### PR TITLE
feat: Add haystack integration

### PR DESCRIPTION
**Pull request by jirispilka**
Documentation for the new Haystack integration.

Marking it as a draft, as the PRs for the Haystack integration page and cookbook examples have not yet been merged.

### CODE REVIEW

File: sources/platform/integrations/ai/haystack.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+print("query='Haystack'")
+for doc in document_store.bm25_retrieval("Haystack", top_k=1):
+    print(doc)
+
+print("Retrieving documents from the document store using vector similarity.")
+for doc in document_store.embedding_retrieval(
+    OpenAITextEmbedder(Secret.from_token(openai_api_key)).run("What is Haystack?")["embedding"], top_k=1
+):
+    print("The closest document to the query 'What is Haystack?':", doc)
+
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:166]:
```suggestion
import os

from haystack import Document, Pipeline
from haystack.components.embedders import OpenAIDocumentEmbedder, OpenAITextEmbedder
from haystack.components.preprocessors import DocumentSplitter
from haystack.components.writers import DocumentWriter
from haystack.document_stores.in_memory import InMemoryDocumentStore

from apify_haystack import ApifyDatasetFromActorCall

os.environ["APIFY_API_TOKEN"] = "YOUR-APIFY-API-TOKEN"
os.environ["OPENAI_API_KEY"] = "YOUR-OPENAI-API-KEY"

document_loader = ApifyDatasetFromActorCall(
    actor_id="apify/website-content-crawler",
    run_input={
        "maxCrawlPages": 3,  # limit the number of pages to crawl
        "startUrls": [{"url": "https://haystack.deepset.ai/"}],
    },
    dataset_mapping_function=lambda item: Document(content=item["text"] or "", meta={"url": item["url"]}),
)

document_store = InMemoryDocumentStore()
print(f"Initialized InMemoryDocumentStore with {document_store.count_documents()} documents")

document_splitter = DocumentSplitter(split_by="word", split_length=150, split_overlap=50)
document_embedder = OpenAIDocumentEmbedder()
document_writer = DocumentWriter(document_store)

pipe = Pipeline()
pipe.add_component("document_loader", document_loader)
pipe.add_component("document_splitter", document_splitter)
pipe.add_component("document_embedder", document_embedder)
pipe.add_component("document_writer", document_writer)

pipe.connect("document_loader", "document_splitter")
pipe.connect("document_splitter", "document_embedder")
pipe.connect("document_embedder", "document_writer")

print("\nCrawling will take some time ...")
print("You can visit https://console.apify.com/actors/runs to monitor the progress\n")

pipe.run({})
print(f"Added {document_store.count_documents()} to vector from Website Content Crawler")

print("\nRetrieving documents from the document store using BM25")
print("query='Haystack'\n")

bm25_retriever = InMemoryBM25Retriever(document_store)

for doc in bm25_retriever.run("Haystack", top_k=1)["documents"]:
  print(doc.content)

print("\nRetrieving documents from the document store using vector similarity.\n")
retrieval_pipe = Pipeline()
retrieval_pipe.add_component("embedder", OpenAITextEmbedder())
retrieval_pipe.add_component("retriever", InMemoryEmbeddingRetriever(document_store, top_k=1))

retrieval_pipe.connect("embedder.embedding", "retriever.query_embedding")

results = retrieval_pipe.run({"embedder":{"text": "What is Haystack?"}})

for doc in results["retriever"]["documents"]:
  print(doc.content)
```

File: sources/platform/integrations/ai/haystack.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+from haystack.utils.auth import Secret
+
+from apify_haystack import ApifyDatasetFromActorCall
+```
+
+Find your [Apify API token](https://console.apify.com/account/integrations) and [OpenAI API key](https://platform.openai.com/account/api-keys) and initialize these into environment variable:
+
+```python
+apify_api_token = "YOUR-APIFY-API-TOKEN"
+openai_api_key = "YOUR-OPENAI-API"
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:43]: I prefer to set it as env variable
```suggestion
import os

os.environ["APIFY_API_TOKEN"] = "YOUR-APIFY-API-TOKEN"
os.environ["OPENAI_API_KEY"] = "YOUR-OPENAI-API-KEY"
```

File: sources/platform/integrations/ai/haystack.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+```python
+document_loader = ApifyDatasetFromActorCall(
+    actor_id="apify/website-content-crawler",
+    run_input={
+        "maxCrawlPages": 3,  # limit the number of pages to crawl
+        "startUrls": [{"url": "https://haystack.deepset.ai/"}],
+    },
+    dataset_mapping_function=lambda item: Document(content=item["text"] or "", meta={"url": item["url"]}),
+    apify_api_token=apify_api_token,
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:56]: We don't need this argument if we set env variables
```suggestion
```

File: sources/platform/integrations/ai/haystack.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+The dataset mapping function is described in more detail in the [Retrieval augmented generation example](https://colab.research.google.com/github/deepset-ai/haystack-cookbook/blob/main/notebooks/apify_haystack_rag.ipynb).
+
+Next, you can utilize the [Haystack pipeline](https://docs.haystack.deepset.ai/docs/pipelines), which helps you chain several processing components together. I
+n this example, we chain the document loader with the document splitter, document embedder, and document writer components.
+
+```python
+document_store = InMemoryDocumentStore()
+
+document_splitter = DocumentSplitter(split_by="word", split_length=150, split_overlap=50)
+document_embedder = OpenAIDocumentEmbedder(api_key=Secret.from_token(openai_api_key))
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:70]:
```suggestion
document_embedder = OpenAIDocumentEmbedder()
```

File: sources/platform/integrations/ai/haystack.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    },
+    dataset_mapping_function=lambda item: Document(content=item["text"] or "", meta={"url": item["url"]}),
+    apify_api_token=apify_api_token,
+)
+```
+
+You can learn more about input parameters on the [Website Content Crawler inputs page](https://apify.com/apify/website-content-crawler/input-schema).
+The dataset mapping function is described in more detail in the [Retrieval augmented generation example](https://colab.research.google.com/github/deepset-ai/haystack-cookbook/blob/main/notebooks/apify_haystack_rag.ipynb).
+
+Next, you can utilize the [Haystack pipeline](https://docs.haystack.deepset.ai/docs/pipelines), which helps you chain several processing components together. I
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:63]:
```suggestion
Next, you can utilize the [Haystack pipeline](https://docs.haystack.deepset.ai/docs/pipelines), which helps you connect several processing components together. I
```

File: sources/platform/integrations/ai/haystack.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+n this example, we chain the document loader with the document splitter, document embedder, and document writer components.
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:64]:
```suggestion
n this example, we connect the document loader with the document splitter, document embedder, and document writer components.
```

File: sources/platform/integrations/ai/haystack.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+pipe.connect("document_loader", "document_splitter")
+pipe.connect("document_splitter", "document_embedder")
+pipe.connect("document_embedder", "document_writer")
+```
+
+Run all the components in the pipeline:
+
+```python
+pipe.run({"document_loader": {}})
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:87]:
```suggestion
pipe.run({})
```

File: sources/platform/integrations/ai/haystack.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+
+After running the pipeline code, you can print the results
+
+```python
+print(f"Added {document_store.count_documents()} to vector from Website Content Crawler")
+
+print("Retrieving documents from the document store using BM25")
+print("query='Haystack'")
+for doc in document_store.bm25_retrieval("Haystack", top_k=1):
+    print(doc)
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:104]:
```suggestion
bm25_retriever = InMemoryBM25Retriever(document_store)
for doc in bm25_retriever.run("Haystack", top_k=1)["documents"]:
  print(doc.content)
```

File: sources/platform/integrations/ai/haystack.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+for doc in document_store.embedding_retrieval(
+    OpenAITextEmbedder(Secret.from_token(openai_api_key)).run("What is Haystack?")["embedding"], top_k=1
+):
+    print("The closest document to the query 'What is Haystack?':", doc)
+
+```
+
+To run it, you can use the following command: `python apify_integration.py`
+
+Haystack is an open-source framework designed for building search systems, Question Answering (QA) systems, and information retrieval applications. It allows developers to create intelligent search functionalities that can understand and process natural language queries.
```
Comment (bilgeyucel) [inline sources/platform/integrations/ai/haystack.md:171]: Is it possible to move this definition to the top of the page? 
```suggestion
[Haystack](https://haystack.deepset.ai/) is an open source framework for building production-ready LLM applications, agents, advanced retrieval-augmented generative pipelines, and state-of-the-art search systems that work intelligently over large document collections. 
```

**Review by bilgeyucel — COMMENTED** [review]
Thank you for the PR @jirispilka! I left my comments

**Comment by jirispilka** [comment]
> Thank you for the PR @jirispilka! I left my comments

@bilgeyucel Thank you for the comments and improvements! I appreciate it.

**Review by bilgeyucel — APPROVED** [review]

File: sources/platform/integrations/ai/haystack.md [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+description: Learn how to integrate Apify with Haystack to work with web data in the Haystack ecosystem.
+sidebar_position: 1
+slug: /integrations/haystack
+---
+
+**Learn how to integrate Apify with Haystack to work with web data in the Haystack ecosystem.**
+
+---
+
+> For more information on Haystack, visit its [documentation](https://docs.haystack.deepset.ai/docs/intro).
```
Comment (TC-MO) [inline sources/platform/integrations/ai/haystack.md:13]: I would suggest removing the blockquote and moving the sentence introducing Haystack to the top.
This way we will: 

* introduce readers to Haystack itself
* present them with link where they can find more information on it (the docs link)
* and after all that we will move onto the example itself

It seems more logical this way

```suggestion
[Haystack](https://haystack.deepset.ai/) is an open source framework for building production-ready LLM applications, agents, advanced retrieval-augmented generative pipelines, and state-of-the-art search systems that work intelligently over large document collections. For more information on Haystack, visit its [documentation](https://docs.haystack.deepset.ai/docs/intro).
```

File: sources/platform/integrations/ai/haystack.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+results = retrieval_pipe.run({"embedder": {"text": "What is Haystack?"}})
+
+for doc in results["retriever"]["documents"]:
+    print(doc.content)
+```
+
+To run it, you can use the following command: `python apify_integration.py`
+
+[Haystack](https://haystack.deepset.ai/) is an open source framework for building production-ready LLM applications, agents, advanced retrieval-augmented generative pipelines, and state-of-the-art search systems that work intelligently over large document collections.
```
Comment (TC-MO) [inline sources/platform/integrations/ai/haystack.md:182]:
```suggestion

```

**Review by TC-MO — APPROVED** [review]
LGTM , left 2 suggestions

**Comment by jirispilka** [comment]
@mtrunkat thanks for merging 👍🏻

**Pull request merged**